src/EventSourcingTests/append_events_with_optimistic_or_exclusive_locks.cs

```diff
@@ -52,13 +53,15 @@ public async Task append_optimistic_happy_path()
         public async Task append_optimistic_sad_path_with_concurrency_issue()
         {
             var streamId = Guid.NewGuid();
-            theSession.Events.StartStream(streamId, new AEvent(), new BEvent());
-            await theSession.SaveChangesAsync();
+            await using var session1 = theStore.LightweightSession(new SessionOptions { Timeout = 1 });
+
+            session1.Events.StartStream(streamId, new AEvent(), new BEvent());
+            await session1.SaveChangesAsync();
 
             // Fetch the expected version
-            await theSession.Events.AppendOptimistic(streamId, new CEvent(), new BEvent());
+            await session1.Events.AppendOptimistic(streamId, new CEvent(), new BEvent());
 
-            await using (var session = theStore.LightweightSession())
+            await using (var session = theStore.LightweightSession(new SessionOptions{Timeout = 1}))
```

Comment:
**QUESTION:** Out of curiosity, why is setting timeout needed here?

Reply:
The test hangs otherwise